src/Features/VisualBasic/Portable/ConvertForEachToFor/VisualBasicConvertForEachToForCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,185 @@
+﻿' Copyright(c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt In the project root For license information
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeActions
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.ConvertForEachToFor
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertForEachToFor
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicConvertForEachToForCodeRefactoringProvider)), [Shared]>
+    Friend Class VisualBasicConvertForEachToForCodeRefactoringProvider
+        Inherits AbstractConvertForEachToForCodeRefactoringProvider(Of ForEachBlockSyntax)
+
+        Protected Overrides ReadOnly Property Title As String
+            Get
+                Return VBFeaturesResources.Convert_For_Each_To_For
+            End Get
+        End Property
+
+        Protected Overrides Function GetForEachStatement(selection As TextSpan, token As SyntaxToken) As ForEachBlockSyntax
+            Dim forEachBlock = token.Parent.FirstAncestorOrSelf(Of ForEachBlockSyntax)()
+            If forEachBlock Is Nothing Then
+                Return Nothing
+            End If
+
+            ' support refactoring only if caret Is on for each statement
+            Dim scope = forEachBlock.ForEachStatement.Span
+            If Not scope.IntersectsWith(selection) Then
+                Return Nothing
+            End If
+
+            ' we don't support colon seperated statements
```

Comment:
really?  why?  if you have the a foreach block that contains a statement that has a :-separated-statement in it... i don't see why we woudln't be able to convert. all we need to do is take all the statements from the foreach and move them into the for, right?

Reply:
might in future iteration. but not in here.

Reply:
tracking issue - https://github.com/dotnet/roslyn/issues/25667